.github/workflows/neon_extra_builds.yml

```diff
@@ -122,7 +125,31 @@ jobs:
         run: cargo build --all --release --timings
 
       - name: Upload the build stats
-        uses: actions/upload-artifact@v3
+        id: upload-stats
+        env:
+          BUCKET: neon-github-public-dev
+          SHA: ${{ github.event.pull_request.head.sha || github.sha }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_DEV }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_KEY_DEV }}
+        run: |
+          REPORT_URL=https://${BUCKET}.s3.amazonaws.com/build-stats/${SHA}/${GITHUB_RUN_ID}/cargo-timing.html
+          aws s3 cp --recursive --only-show-errors ./target/cargo-timings/ "s3://${BUCKET}/build-stats/${SHA}/${GITHUB_RUN_ID}/"
+          echo "report-url=${REPORT_URL}" >> $GITHUB_OUTPUT
```

Comment:
Not sure I follow the question, but let me explain what happens here: 
* we have a public S3 bucket that is able to display its contents, anyone can go to `https://${BUCKET}.s3.amazonaws.com` , add some relative path in S3 and get the file returned.

Allure manipulates (and exposes) files under `reports/` subpath, I'm here adding another one, `build_stats/` and put there the HTMLs `cargo --timings` produces.

Now, I have to expose that html url (REPORT_URL) as a GH commit status (same way Allure reports do that), but that uses `script: |` for that and I'm not sure it's mixable with plain bash commands here, so instead of doing that directly, I write the URL into a shared var , `report-url`, that's exposed in the task output.

Then access this var and do `createCommitStatus` on its value, to display the report file URL on a commit.